src/ttt/ui.clj

```diff
@@ -0,0 +1,31 @@
+(ns ttt.ui
+    (:require [clojure.math.numeric-tower :as math]))
+
+(defn- add-padding-to-cells
+  [row]
+  (clojure.string/join " | " row))
+
+(defn- print-first-player-message
+  [[player1 player2]]
+  (println "Player 1 is" (:marker player1)))
+
+(defn print-player-intructions
+  []
+  (println "Enter a number between 0-8 to select a spot."))
+
+(defn print-start-game-message
+  [players]
+  (println "Let's play tic-tac-toe.")
+  (print-first-player-message players)
+  (print-player-intructions))
+
+(defn print-board
+  [board]
+  (let [size (math/sqrt (count board))]
+    (let [rows (into [] (map add-padding-to-cells (partition size board)))
+          divider (str "\n" (clojure.string/join (repeat (count (first rows)) "-")) "\n")]
```

Comment:
I'm also a bit confused why `divider` isn't within its own list structure.  Is this `let` binding `rows` and `divider`, or just `rows`?

If it's binding both, I was thinking it would look more like this?

```clojure
(let [rows (into [] (map add-padding-to-cells (partition size board)))]
     [divider (str "\n"
                   (clojure.string/join
                      (repeat (count (first rows)) "-")
                      "\n"))])
  ...)
```

Reply:
this is a clojure syntax thing

https://clojuredocs.org/clojure.core/let

```
(let [name1 thing1-to-bind-to
      name2 thing2-to-bind-to])
```
I think it's because all params go into the []